funsor/adjoint.py

```diff
@@ -225,9 +226,9 @@ def _scatter(src, res, subs):
                 index.append(int(v.data))
             else:
                 # Permute and expand v.data to end up at new_dims.
-                assert isinstance(v, Tensor)
+                assert is_tensor(v)
```

Comment:
As discussed offline, I think it is better to refactor `Tensor+Array` before refactoring `Gaussian`.